src/skydb.js

```diff
@@ -0,0 +1,75 @@
+"use strict";
+
+const FormData = require("form-data");
+const fs = require("fs");
+const p = require("path");
+
+const { Upload } = require("tus-js-client");
+
+const { buildRequestHeaders, SkynetClient } = require("./client");
+const { defaultOptions, getFileMimeType, makeUrl, walkDirectory, uriSkynetPrefix } = require("./utils");
+
+/**
+ * The tus chunk size is (4MiB - encryptionOverhead) * dataPieces, set in skyd.
+ */
+const TUS_CHUNK_SIZE = (1 << 22) * 10;
+
+/**
+ * The retry delays, in ms. Data is stored in skyd for up to 20 minutes, so the
+ * total delays should not exceed that length of time.
+ */
+const DEFAULT_TUS_RETRY_DELAYS = [0, 5000, 15000, 60000, 300000, 600000];
+
+const defaultUploadOptions = {
+  ...defaultOptions("/skynet/skyfile"),
+  endpointLargeUpload: "/skynet/tus",
+
+  portalFileFieldname: "file",
+  portalDirectoryFileFieldname: "files[]",
+  customFilename: "",
+  customDirname: "",
+  dryRun: false,
+  errorPages: undefined,
+  largeFileSize: TUS_CHUNK_SIZE,
+  retryDelays: DEFAULT_TUS_RETRY_DELAYS,
+  tryFiles: undefined,
+};
+
+const nodejs_db_setJSON = async function (privateKey, dataKey, data, customOptions = {}) {
```

Comment:
We should try to be consistent and use the syntax we use in other files, e.g.:

```js
SkynetClient.prototype.uploadFile = async function (path, customOptions = {}) {
```

Reply:
Hello M-Cup,

Yes, I tried to do that with "prototype" at the beginning, but everything went wrong.
Further failures followed to integrate it as with "Skynet-Js" until I found one on this integration path!
What I tried with "prototype" was able to get the command:
"client.db.setJSON"

If you know how I can do the command in a "prototype" build, then gladly.

The false tests only have
"client.setJSON" or "client.db_setjson"
had no "db".

I also got it with "Pretier". Error is gone.
And I have expanded "setJSON" and refined the original faithful to "skynet-Js"! But not quite finished yet.